### PR TITLE
Add RESTful Api

### PR DESCRIPTION
**Pull request by Alabi12**
# Summary
**The following were the tasks added to the Blog app**

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

**Review by Reem-lab — COMMENTED** [review]
Hi @Alabi12, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

❌❌❌
## Invalid Code Review Request

You have submitted a project, that the milestone is made just by you since it should be pair programming but you should do it with your partner please let your partner implement it with you 🤩🙏🏻💯✨🔥✅

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/api/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+class Api::CommentsController < Api::ApplicationController
+  def index
+    @comments = Post.find(params[:post_id]).comments
+    render json: @comments
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = Comment.create!(params.require(:comment).permit(:text)
+     .merge(author_id: params[:user_id], posts_id: @post.id))
```
Comment (DammyShittu) [inline app/controllers/api/comments_controller.rb]: - [ ] I cannot create a comment. This should be `post_id`, not `posts_id`.

  ```
  .merge(..., post_id: @post.id))
  ```

  uiffk

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Alabi12 and @Ibnballo1,

## Highlights
✔️ Creation of API endpoints

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @Alabi12 and @Ibnballo1,

## APPROVED 🟢

## Highlights
✔️ Comments can be created using the API endpoint

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**